flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointRetentionPolicy.java

```diff
@@ -24,6 +24,9 @@
 @Internal
 public enum CheckpointRetentionPolicy {
 
+    /** Full Checkpoints should be retained on cancellation and failure. */
+    FULL_RETAIN_ON_CANCELLATION,
```

Comment:
Whether the checkpoint is full or not doesn't fit with the `CheckpointRetentionPolicy`. It should be an independent axis.

If the checkpoint is full or incremental it should be parametrised independently of the retention policy. Assuming that we want to expose retention to users - and I don't see why we should be doing this. Is there some particular reason why have you implemented it like this @leletan ? Such exposure makes this more complicated for us to maintain code in the future and it also makes it more complicated for the users. To me it sounds like this retention policy should come from the `CheckpointCoordinatorConfiguration` and be the same for manually and automatically triggered checkpoints.

Reply:
Good catch @pnowojski. This was one of the place I was not very confident with. As a hindsight, totally agree it was a bad design.

The reason why I did that was as following:
I was not sure if a CheckpointProperty will be valid if it is too far away from [these](https://github.com/apache/flink/blob/d8a4304b892412eab4c5c19b5deb84166943d3bb/flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointProperties.java#L246) pre-configured ones so I was thinking of creating one as close to an existing one as possible. Meanwhile the public method to create a pre-configured CheckpointProperty was through [this](https://github.com/apache/flink/blob/d8a4304b892412eab4c5c19b5deb84166943d3bb/flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointProperties.java#L349) method with a `CheckpointRetentionPolicy`.

Meanwhile I would love to hear your thoughts 
1. What checkpoint configurations we should expose in the restful API? Are there more general use cases requesting other configurations for checkpoint manual triggering (other than ours)?
2. Or would it be better just to start with our specific use case - always only trigger a full checkpoint, i.e.  reuse all `CheckpointProperty` and only replace `CheckpointType` with [`FULL_CHECKPOINT `](https://github.com/apache/flink/blob/d8a4304b892412eab4c5c19b5deb84166943d3bb/flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointType.java#L30)

Reply:
I could see use cases for manually triggering incremental checkpoints as well, so I would suggest to full/incremental parameter. We would need to decouple REST API constants from the internal classes and add a translation layer for that. It should be probably something similar to `org.apache.flink.core.execution.SavepointFormatType`. `CheckpointType` with two supported values `INCREMENTAL` and `FULL`. 

As for the `CheckpointProperties` class, probably you should copy instead of passing `CheckpointProperties` in `org.apache.flink.runtime.checkpoint.CheckpointCoordinator#triggerCheckpoint(org.apache.flink.runtime.checkpoint.CheckpointProperties)`, you should just pass `org.apache.flink.runtime.checkpoint.CheckpointType` and inside the `triggerCheckpoint` construct new `CheckpointProperties` by copying `org.apache.flink.runtime.checkpoint.CheckpointCoordinator#checkpointProperties`. Something like that:
```

    public CompletableFuture<CompletedCheckpoint> triggerCheckpoint(CheckpointType checkpointType) {
        CheckpointProperties properties =
                new CheckpointProperties(
                        checkpointProperties.forceCheckpoint(),
                        checkpointType,
                        checkpointProperties.discardOnSubsumed(),
                        checkpointProperties.discardOnJobFinished(),
                        checkpointProperties.discardOnJobCancelled(),
                        checkpointProperties.discardOnJobFailed(),
                        checkpointProperties.discardOnJobSuspended(),
                        checkpointProperties.isUnclaimed());
        return triggerCheckpointFromCheckpointThread(properties, null, false);
    }
```